src/utils/cache.ts

```diff
@@ -0,0 +1,79 @@
+import axios from "axios"
+
+const ONE_HOUR = 1000 * 60 * 60
+
+export interface CacheStorage<T> {
+  value: T
+  timestamp: number
+}
+
+const emptyCacheItemStorage: CacheStorage<string> = {
+  value: "",
+  timestamp: 0,
+}
+
+const stringifyEmptyCacheItemStorage = JSON.stringify(emptyCacheItemStorage)
+
+/**
+ * Write inside the local storage the value returned by the API call for a given key
+ * @param {string} key stored in the local storage
+ * @param {T} value for a given key to write in the local storage
+ */
+export const writeToCache = <T>(key: string, value: T): void => {
+  const cacheStorageItem: CacheStorage<T> = {
+    value,
+    timestamp: new Date().getTime(),
+  }
+  localStorage.setItem(key, JSON.stringify(cacheStorageItem))
+}
+
+/**
+ * Retrieve the value stored in the local storage for a given key
+ * If the value is not defined or the key does not exist, return a default CacheStorage
+ * with an empty value and timestamp set to 0
+ * @see CacheStorage
+ * @param {string} key stored in the local storage
+ * @returns
+ */
+const readFromCache = <T>(key: string): CacheStorage<T> => {
+  return (
+    JSON.parse(localStorage.getItem(key) || stringifyEmptyCacheItemStorage) ||
+    emptyCacheItemStorage
```

Comment:
Yes I overdid it.
Moreover, I added a type assertion because JSON.parse returns an any type from the typescript compiler. I force to tell Typescript that the type is `CacheStorage<T>`